### ISSUE TITLE
Militia missile argosys never fire missiles, and instead just fly away

### ISSUE THREAD

**Issue by Bladewood** [endless-sky/endless-sky#1871]
There is always this one argosy that just flies away, contributing nothing to the battle (other than shooting down random incoming missiles).

I can tell that it's a missile argosy because it is the only argosy variant that uses atomic engines.

On an unrelated note, what is the plural of "argosy"?

**Comment by Wrzlprnft** [comment]
Maybe the variant needs the gunpoints declared additionally? It has turrets declared, which may wipe everything regarding the weapon hardpoints?

**Comment by kozbot** [comment]
I do believe it clears the hardpoints first, if it sees any hardpoints it makes a new Armament overwriting the old one.  Ref Ship.cpp Line#95-122

**Comment by Wrzlprnft** [comment]
The questionmarks are a result of me going by train and just having mobile.
Can someone please test this by adding gun hardpoints?

**Comment by Elyssaen** [comment]
Just set up a test against an Argosy (Missile) and it used the normal missile boat AI – kept distance while battering me with Meteor Missiles. Bladewood, are you sure it hadn't run out of missiles by the time you noticed it? I'll switch to my 'original' .exe to check, and look at **Ship.cpp** too, but... it's behaving for me.

**Comment by KiLEdEnNis** [comment]
and shouldn't the plural for Argosy be either Argosies (legit) or Argosa (sounds funny)?

**Comment by Bladewood** [comment]
@Wrzlprnft According to the [wiki](https://github.com/endless-sky/endless-sky/wiki/CreatingShips), and to my own experiments as well, turret hardpoints, gun hardpoints, and engine hardpoints are independent of each other when it comes to variants.
> In this case, all four guns are the same type (heavy lasers), so there is no need to specify their order, but the order of the turrets [2x heavy laser turrets, 1x heavy anti-missile turret] must be specified.

@Elyssaen It probably is their frugal attribute, as you say.

**Comment by KiLEdEnNis** [comment]
frugal only lets them shoot when they are outnumbered or below... 90% hull?

**Comment by Elyssaen** [comment]
50% shielding. Well, technically if the combined shield and hull percentages goes below 150%. And outnumbered is I think based on total ship cost.

Frugal missile ships... I mostly agree they shouldn't happen, but there are cases where it makes sense. If your non-ammo-using allies will easily win, why bother firing? Perhaps rename this issue (or remake for a clean slate) to discuss this though, as it's now a more general issue. : )

**Comment by Bladewood** [comment]
As you command.

**Comment by KiLEdEnNis** [comment]
I prefer Argosa :3

**Comment by endless-sky** [comment]
It's "argosies." Just look it up in [Merriam Webster](https://www.merriam-webster.com/dictionary/argosy):

> a large ship; especially a large merchant ship (three of your argosies are … come to harbor — Shakespeare)

**Comment by Bladewood** [comment]
Wait, that's actually a word?!

**Comment by KiLEdEnNis** [comment]
I thought it would be Argosies, but Argosa sounds more fun :(

**Comment by geojak** [comment]
Is this behaviour still in game? The linked issue above has been closed already

**Comment by Amazinite** [comment]
The linked issue was closed because it was a duplicate of other issues.

**Comment by warp-core** [comment]
Is this a problem? What's the desired solution? Once #6822 is done, would that enable this to be solved? The missile Argosies could have their missile launchers specifically set to always fire?

**Comment by Amazinite** [comment]
Frugal ships shouldn't follow missile boat behavior unless they have decided to fire.
